module/default.nix

```diff
@@ -10,29 +10,39 @@ let
   ageBin = if cfg.isRage then "${cfg.pkg}/bin/rage" else "${cfg.pkg}/bin/age";
 
   runtimeDecryptPath = path: runtimeDecryptFolder + "/" + path;
-  encryptedPath = path: cfg.folder + "/" + path + ".age";
 
   identities = builtins.concatStringsSep " " (map (path: "-i ${path}") cfg.identityPaths);
-  createLinks = runtimepath: symlinks: builtins.concatStringsSep "\n" ((map (link: "ln -sf ${runtimepath} ${link}")) symlinks);
 
-  decryptSecret = name: { path, symlinks, mode, owner, group, ... }:
+  createLinks = runtimepath: symlinks: builtins.concatStringsSep "\n" ((map (dest: ''
+    mkdir -p $(dirname ${dest})
+    ln -sf ${runtimepath} ${dest}
+  '')) symlinks);
+  createCopies = runtimepath: copies: builtins.concatStringsSep "\n" ((map (dest: ''
+    mkdir -p $(dirname ${dest})
+    cp -f ${runtimepath} ${dest}
+  '')) copies);
+
+  decryptSecret = name: { source, path, symlinks, copies, mode, owner, group, ... }:
     let
       runtimepath = runtimeDecryptPath path;
-      encryptpath = encryptedPath path;
-      links = createLinks runtimepath symlinks;
+      linksCmds = createLinks runtimepath symlinks;
+      copiesCmds = createCopies runtimepath copies;
```

Comment:
Copies should be during an activationScript rather than the oneshot service. Otherwise unnecessarily overwriting on each systemd service.